drivers/ethernet/eth_stm32_hal.c

```diff
@@ -98,43 +203,78 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 	ETH_HandleTypeDef *heth;
 	uint8_t *dma_buffer;
 	int res;
-	uint16_t total_len;
+	uint32_t u32_total_len;
 	__IO ETH_DMADescTypeDef *dma_tx_desc;
+	HAL_StatusTypeDef hal_ret = HAL_OK;
 
 	__ASSERT_NO_MSG(pkt != NULL);
 	__ASSERT_NO_MSG(pkt->frags != NULL);
 	__ASSERT_NO_MSG(dev != NULL);
 	__ASSERT_NO_MSG(dev_data != NULL);
 
-	heth = &dev_data->heth;
+	heth = &(dev_data->heth);
 
 	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
 
-	total_len = net_pkt_get_len(pkt);
-	if (total_len > ETH_TX_BUF_SIZE) {
+	u32_total_len = net_pkt_get_len(pkt);
```

Comment:
"hungarian notation" ?

Reply:
> "hungarian notation" ?

Yes, it is. Changed back to `total_len`.